### ISSUE TITLE
Hangup in timedWait

### ISSUE THREAD

**Issue by nambirajang** [WebPlatformForEmbedded/WPEWebKit#525]
We are seeing hang in timedWait API during Youtube video playback.

it seems like https://bugs.webkit.org/show_bug.cgi?id=183348 .
Can you please confirm back-porting of below related commits from webkit master to wpe-20170728 will resolve this issue.

https://github.com/WebKit/webkit/commit/a69f99f3d63b06b16510e7eafdcf892a45952a23#diff-174804561819a89edd935f2390bf46ed
https://github.com/WebKit/webkit/commit/c47296d6fefbe386c6bc78edc3bbb05b5952ff98#diff-174804561819a89edd935f2390bf46ed
https://github.com/WebKit/webkit/commit/f0627f5287186ec975c3439e9bcfe119493adc95#diff-174804561819a89edd935f2390bf46ed​

[timedwait-1.txt](https://github.com/WebPlatformForEmbedded/WPEWebKit/files/2283868/timedwait-1.txt)

**Comment by carlosgcampos** [comment]
The hang is in sendSync, when the WebProcess tries to ensure a connection to the network process, sending the message GetNetworkProcessConnection to the UI process. What we need to understand here is why the UI process doesn't reply to the message. It would be interesting to get the backtrace of the UI process too. I don't think backporting those patches would help, since they are focused on what to do when sending the sync message fails, but in this case, it's not failing, the message is sent and we keep waiting forever for the response. The key point here is why the UI process doesn't reply.

**Comment by nambirajang** [comment]
Hi @carlosgcampos, Thanks for the clarification.
I remember we had this issue very long back. https://metrological.atlassian.net/browse/COMCAST-36.
We found that socket descriptor has been removed from event loop due to less priority compared to  Runloop descriptor.

``void GSocketMonitor::start(GSocket* socket, GIOCondition condition, RunLoop& runLoop, std::function<gboolean (GIOCondition)>&& callback)
{
..
g_source_set_priority(m_source.get(), G_PRIORITY_HIGH + 30);
``
In the latest code, we are setting the socket priority same as RunLoopSourcePriority.
(i.e. https://bugs.webkit.org/show_bug.cgi?id=172160)

We started using this latest code recently. I will monitor this issue in upcoming release builds and close this issue if it is not reproducible.
